crates/nu-parser/src/parse_keywords.rs

```diff
@@ -40,6 +40,8 @@ use crate::{
 /// These parser keywords can be aliased
 pub const ALIASABLE_PARSER_KEYWORDS: &[&[u8]] = &[b"overlay hide", b"overlay new", b"overlay use"];
 
+pub const RESERVED_KEYWORDS: [&str; 3] = ["in", "nu", "env"];
```

Comment:
Thanks for factoring this into a common place!

Maybe worth naming separate from the "keyword-keywords"

```suggestion
pub const RESERVED_VARIABLE_NAMES: [&str; 3] = ["in", "nu", "env"];
```